libraries/Microsoft.Bot.Builder.Dialogs/InputRecognizer.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    public class InputRecognizer 
+    {
+        /// <summary>
+        /// Gets or sets id of the recognizer.
+        /// </summary>
+        /// <value>Id.</value>
+        [JsonProperty("id")]
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Runs current DialogContext.TurnContext.Activity through a recognizer and returns a generic recognizer result.
+        /// </summary>
+        /// <param name="dialogContext">Dialog context.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>Analysis of utterance.</returns>
+        public virtual Task<RecognizerResult> RecognizeAsync(DialogContext dialogContext, CancellationToken cancellationToken = default)
+        {
```

Comment:
I know you are not that fond of the => pattern, but either we should disable that rule or follow the pattern.  I suspect you have a bunch of style cop warnings...